### ISSUE TITLE
MERGE INTO statement handles not returning total count of rows affected

### ISSUE THREAD

**Issue by dr-kd** [bucardo/dbdpg#118]
I think that this is a bug where DBD::Pg isn't handling the MERGE command tag

Postgres 15 supports MERGE INTO but the number of records coming back from the `$sth->execute` doesn't reflect this.

The [pg merge documentation](https://www.postgresql.org/docs/current/sql-merge.html) says:

> On successful completion, a MERGE command returns a command tag of the form `MERGE total_count` 

This is similar to the docs for INSERT.  

I execute the following  sql using psql and get the following results:

```
create temporary table t (id bigint, msg text);
CREATE TABLE
insert into t (id, msg) values (1, 'here you go');
INSERT 0 1
with quack (id, msg) as ( select 2, 'another' union select 1, 'edit original edit2' 
    )  merge into t using (select * from quack) as d 
     on (t.id = d.id)
     when matched and t.id = d.id
      then update set msg = d.msg
     when not matched 
      then insert values(d.id, d.msg);
MERGE 2
```

Then if I run the same with the following perl script:

``
`#!/usr/bin/env perl
use warnings;
use strict;
use DBI;
use Data::Dumper;
my @sql = (
q|create temporary table t (id bigint, msg text)|,
q|insert into t (id, msg) values (1, 'here you go')|,
q|with quack (id, msg) as ( select 2, 'another' union select 1, 'edit original' 
    )  merge into t using (select * from quack) as d 
     on (t.id = d.id)
     when matched and t.id = d.id
      then update set msg = d.msg
     when not matched 
      then insert values(d.id, d.msg)|,
    );

my $dbh = DBI->connect('dbi:Pg:host=MYHOST;dbname=MYDB', 'user', 'pass');

for my $s (@sql) {
    my $h = $dbh->prepare($s);
    my @res = $h->execute;
    print Dumper { sql => $s, result => \@res};
}

```
I would expect the result to contain the number of records handled 

Omitting all but the last (MERGE INTO) query:

```
$VAR1 = {
          'sql' => 'with quack (id, msg) as ( select 2, \'another\' union select 1, \'edit original edit2\' 
    )  merge into t using (select * from quack) as d 
     on (t.id = d.id)
     when matched and t.id = d.id
      then update set msg = d.msg
     when not matched 
      then insert values(d.id, d.msg)',
          'result' => [
                        '0E0'
                      ]
        };
```

**Comment by singingfish** [comment]
capitalise the sql keywords here: 	       with s as (                                                                                                                                                                 
        select prodcode, string_agg(plant,' ' order by plant)  as available_plants                                                                                                  
                  from state_item_available where available_redirected > 0                                                                                                          
                 group by prodcode                                                                                                                                                  
        )                                                                                                                                                                           
            merge into state_plants_redirected d using                                                                                                                              
                (select * from s) as s                                                                                                                                              
                on (s.prodcode = d.prodcode)                                                                                                                                        
            when matched AND d.available_plants != s.available_plants then                                                                                                          
                update set                                                                                                                                                          
                    available_plants = s.available_plants                                                                                                                           
            when not matched then                                                                                                                                                   
                insert                                                                                                                                                              
                values (s.prodcode, s.available_plants)

**Comment by turnstep** [comment]
Thanks for noticing this: applied a fix in c843c4de14e341786b51f2a87d4738510c556ecf

**Comment by turnstep** [comment]
Fixed as of version 3.17.0
